klient/conf/config.go

```diff
@@ -33,14 +34,33 @@ var DefaultClusterContext = ""
 // New returns Kubernetes configuration value of type *rest.Config.
 // filename is kubeconfig file
 func New(fileName string) (*rest.Config, error) {
-	// if filename is not provided assume in-cluster-config
+	var resolvedKubeConfigFile string
+	context := ResolveClusterContext()
+
+	// resolve the kubeconfig file
 	if fileName == "" {
-		return rest.InClusterConfig()
+		resolvedKubeConfigFile = ResolveKubeConfigFile()
+	} else {
+		resolvedKubeConfigFile = fileName
+	}
+
+	// if resolvedKubeConfigFile is still empty, assume in-cluster config
+	if resolvedKubeConfigFile == "" {
+		if context == "" {
+			return rest.InClusterConfig()
+		}
+		// if in-cluster can't use the --context flag
+		return nil, errors.New("cannot use --context flag without a kubeconfig file")
```

Comment:
I would reword the error to just mention "cluster context" instead of "--context flag" since context can come in programmatically at this point in the code, not necessarily via flag.

Reply:
The argument is `--kube-context` right? Can you please update the name of the arg as well

Reply:
@harshanarayana , no as @vladimirvivien said in the previous comment, we switched back to `--context` to be consistent with other Kubernetes CLI. For example `kubectl` uses `--context`.

Reply:
Sorry, I totally missed that part of the discussion. @maruina Thanks